example_client_from_context_dbsql_test.go

```diff
@@ -0,0 +1,88 @@
+package river_test
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"log/slog"
+	"time"
+
+	_ "github.com/jackc/pgx/v5/stdlib"
+
+	"github.com/riverqueue/river"
+	"github.com/riverqueue/river/internal/riverinternaltest"
+	"github.com/riverqueue/river/riverdriver/riverdatabasesql"
+	"github.com/riverqueue/river/rivershared/util/slogutil"
+)
+
+type ContextClientSQLArgs struct{}
+
+func (args ContextClientSQLArgs) Kind() string { return "ContextClientSQLWorker" }
+
+type ContextClientSQLWorker struct {
+	river.WorkerDefaults[ContextClientSQLArgs]
+}
+
+func (w *ContextClientSQLWorker) Work(ctx context.Context, job *river.Job[ContextClientSQLArgs]) error {
+	client := river.ClientFromContext[*sql.Tx](ctx)
+	if client == nil {
+		fmt.Println("client not found in context")
+		return errors.New("client not found in context")
+	}
+
+	fmt.Printf("client found in context, id=%s\n", client.ID())
+	return nil
+}
+
+// ExampleClientFromContext_databaseSQL demonstrates how to extract the River
+// client from the worker context when using the [database/sql] driver.
+// ([github.com/riverqueue/river/riverdriver/riverdatabasesql])
+func ExampleClientFromContext_databaseSQL() {
```

Comment:
These examples are named so that they can be correctly associated with the `ClientFromContext` function. The pkgsite docs will group them together underneath the function's documentation.

https://pkg.go.dev/github.com/fluhus/godoc-tricks#Examples